Torch Jobs

Raptor1628 opened the pull request:
<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->

Adds job datums, outfits, and cards to the jobs file, along with a few
extra stuff required to make it work. Does not affect the player side of things so no changelog.

Casper3667 committed:
made limbs printable in the organ printer

atlantiscze committed:
Significantly increases meteor difficulty
- Note: Apparently people think it's not dangerous enough. It is now. Have fun.
- Tweak: Waves come 2x more often (4 minutes -> 2 minutes)
- Tweak: Maximal severity increased to 40. New group of meteors (Cataclysm) added that triggers at 20+ and is even more "fun" than armageddon, with about 40% chance of a tunguska or EMP meteor. In general, this group is one of the most dangerous things for a shield generator.
- Tweak: Major random events fire even more often in meteor mode(about 3x more often than default)
- Tweak: Escalation probability increased (30% -> 50%).
- Tweak: Some stuff moved from defines to variables, should allow for in-round tweaking/admin fun if necessary.
- Tweak: Failsafe updated accordingly, now forces severity 15 (pretty nasty, but with some luck, survivable.)
- Tweak: Reduced players needed to start back to 5. If people want the mode, let them, i guess.
- Tweak: Ore drops from meteors reduced accross the board. Meteor-shield mining is still very profitable (and dangerous), but not so massively.
- Added: Admin notifications, that can be enabled by enabling one var. Primarily used for debugging purposes, but i thought someone may find it interesting.
- Added: Golden and silver meteors, which drop a little of silver and gold ore on impact. Relatively harmless in comparison to other meteor types.
- Added: Supermatter shard meteor type. Relatively harmless against structures (comparable to big meteor combined with EMP meteor), but insanely powerful against shield generators. Very rare, and occurs only in late-game meteor gamemode.
- Tweak: Highly buffed the random meteor storm event, both in terms of duration and meteor count. In general meteor storms are once again pretty dangerous.
- Bugfix: Fixed shuttle_delay variable on gamemode doing nothing, it now correctly multiplies shuttle transit time.

May the gods be with you. Good luck.

babydoll committed:
Fixes sheet snatcher not collecting items on click.

babydoll committed:
Changelog

babydoll committed:
Adds color changeable hoodies to loadout menu
Actually fixes black jumpskirt icon
Changelog for crew manifest

Raptor1628 committed:
Torch Jobs

Adds job datums, outfits, and cards to the jobs file, along with a few
extra stuff required to make it work.

Raptor1628 commented:
This compiled correctly, so not sure what's up.

PsiOmegaDelta commented:
You added two << shifts. Will have to update .travis.yml accordingly.

Raptor1628 commented:
Do I have to do that? If so, how?

thasc commented:
This guy [here](https://github.com/Baystation12/Baystation12/blob/dev/.travis.yml#L58) - change 771 to 773 in the two places it's present on that line, and you'll be set.

Raptor1628 committed:
Travis

Raptor1628 commented:
Well, there it goes again

thasc commented:
Different error this time at least. I'll have a look.

thasc commented:
Ok, it's failing two tests now.

First test: In `torch_jobs.dm`, `/var/const/access_senmed` and `/var/const/access_senadv` both have the value 91. I think `/var/const/access_senadv` wants the value 92 instead - i.e. they all need to be unique and apparently incrementing. No idea how any of that system works, but that's why it's moaning.

Second test: Still in `torch_jobs.dm`, some different outfit declarations share name values, which apparently isn't allowed (no idea how that system works either, but them's the rules). So the following need unique names:

`/decl/.../research/assist/testsubject` has the same name value ("Custodian - Torch") as `/decl/.../research/assist/janitor`

`/decl/.../research/scientist/psych` has the same name value ("Psychiatrist - Torch") as `/decl/.../medical/counselor/psychiatrist`

`/decl/.../engineering/roboticist` has the same name value ("Roboticist") as `/decl/.../science/roboticist`

`/decl/.../passenger`, `/decl/.../job/torch`, `/decl/.../job/torch/crew` and `/decl/.../torch/crew/command` all have the same name value (probably the default value, since these guys all seem to be superclasses and might not have been given a name).

Raptor1628 committed:
Fixes

Raptor1628 commented:
Yeah, saw that one myself. Got a fix up though. Thanks for the continuing help. Getting so damn close.

PsiOmegaDelta committed:
Mindless mobs no longer crash the cryopod.

However, it does raise a whole lot of other concerns...
Fixes #15019. Fixes #15021.

PsiOmegaDelta committed:
Updates changelog.

LorenLuke committed:
Tweaks grab behaviors when throwing and checks adjacency before grab actions. Fixes being able to 100% resist out of grab_kill (so changelings can actually absorb now).

PsiOmegaDelta committed:
Merge pull request #15040 from PsiOmegaDelta/161211-Changelog

Updates changelog.

comma committed:
Merge pull request #15039 from PsiOmegaDelta/161211-SleepNowSleepForever

Mindless mobs no longer crash the cryopod.

comma committed:
Merge pull request #15031 from sabiram/sheetsnatcher

Fixes sheet snatcher not collecting items on click

Kasuobes commented:
Hrm, everyone spawning w/ EC uniforms... lemme see what sort of black magic can be hacked in.

babydoll committed:
Loads of adjustments

Removes grey hoodie and black hoodie icons
Retains black hoodie as a recolored subtype for compatibility
Enforces loadout menu naming scheme standard

Raptor1628 commented:
I mean, it'd be nice if branch determined uniform, or you could set up alt-titles to do outfits, but this works alright, I'll be setting custom loadout to allow the other uniforms and such, and it'll be up to folks ICly to sort it out.

atlantiscze committed:
Quick-Fix for shield lag problems

- Hopefully fixes #15050
- Shields will no longer spark when shut down due to power outage, only segments that fail due to damage/diffuser effect do.

Raptor1628 commented:
So uh, is there anything else this needs?

PsiOmegaDelta commented on code/game/jobs/jobs.dm:
```diff
@@ -6,6 +6,8 @@ var/const/CIV               =(1<<4)
 var/const/CRG               =(1<<6)
 var/const/MSC               =(1<<7)
+var/const/SRV				=(1<<8)
+var/const/SUP				=(1<<9)
```
Mixed indentation types. Pre-existing variants use spaces, you're using tabs. GinjaNinja32 would never let me pass that.

Raptor1628 replied:
Meh, there's gotta be an easier way to tell with is used than manually checking each one. I'll get to this after I'm done with this set of finals.

PsiOmegaDelta reviewed (changes requested):
No major issues noted, only the minor one commented upon.
However, I'll be doing some code shuffling once this is merged.

Raptor1628 commented:
So far as shuffling, I already know I'll have to move the access vars over to the main file so that my lockers  work.

PsiOmegaDelta commented:
My plan is to move as much as possible away from the main files as well as split the Torch files down into smaller components, because files with 4 digit  number of lines are evil.

SiegDerMaus committed:
Makes Zip Guns Buildable (#14879)

* Makes Zip Guns Buildable

PsiOmegaDelta committed:
Merge pull request #14821 from Casper3667/organprinter

allows printing of limbs

PsiOmegaDelta committed:
Merge pull request #15028 from atlantiscze/2016-12-08-ExtinctionLevelEvent

Significantly increases meteor difficulty

PsiOmegaDelta committed:
Merge pull request #15041 from LorenLuke/Grab-Tweaks

Grab Changes

PsiOmegaDelta committed:
Merge pull request #15042 from sabiram/hoodiesandsuch

Color selectable hoodies, actually fixes black jumpskirt icon

PsiOmegaDelta committed:
Merge pull request #15054 from atlantiscze/2016-12-12-QuickFix

Quick-Fix for shield lag problems

PsiOmegaDelta committed:
Updates changelog.

PsiOmegaDelta committed:
Adjusts the angle bracket count.

PsiOmegaDelta committed:
Merge pull request #15060 from PsiOmegaDelta/161213-Changelog

Updates changelog.

Raptor1628 commented:
As long as uniforms and equipment aren't moved to map datums, organizing is fine.

Raptor1628 committed:
Requested changes and fixes merge conflicts

Raptor1628 committed:
Fixes merge conflicts

Raptor1628 committed:
Travis

Raptor1628 committed:
Merge branch 'dev' of https://github.com/Raptor1628/Baystation12 into dev

Raptor1628 commented:
This is going to need a squash, that's still possible on your end right?

Kasuobes commented:
You still have less commits than files changed, meh.

Raptor1628 commented:
I have no idea what it did, and I'm probably going to have to just redo the PR -_-
